jgt_common/assert_.py

```diff
@@ -1,132 +1,58 @@
 """Convenience functions for doing asserts with helpful names and helpful messages."""
-from jgt_common import percent_diff
-from . import format_if as _format_if
 
-try:
-    from math import isclose as _isclose
-except ImportError:
-    # noqa E501 From: https://stackoverflow.com/questions/5595425/what-is-the-best-way-to-compare-floats-for-almost-equality-in-python/5595453
-    def _isclose(a, b, rel_tol=1e-09, abs_tol=0.0):
-        return abs(a - b) <= max(rel_tol * max(abs(a), abs(b)), abs_tol)
+# NOTE TO IMPLEMENTORS:
+# This module depends on the contracts described in our sibling module ``check``.
+# In the spirit of DRY, please see that module for details.
 
 
-def _msg_concat(prefix, body):
-    """Join with a space if prefix isn't empty."""
-    return "{} {}".format(prefix, body) if prefix else body
+from functools import wraps as _wraps
+from inspect import getmodule as _getmodule
+from inspect import getmembers as _getmembers
+from inspect import isfunction as _isfunction
 
+from . import check as _check
 
-def not_eq(expected, actual, msg=""):
-    """Assert the values to not be equal."""
-    assert expected != actual, _msg_concat(
-        msg, "Expected '{}' to be not equal to actual '{}'".format(expected, actual)
-    )
 
+# Used internally, but there is no reason to prevent it from being used externally.
+def assert_if_truthy(fun):
+    """(Decorator) Assert if fun returns a truthy value (error indicator)."""
 
-def eq(expected, actual, msg=""):
-    """Assert the values are equal."""
-    assert expected == actual, _msg_concat(
-        msg, "Expected '{}' == actual '{}'".format(expected, actual)
-    )
+    # NOTE: Not using ``wrapt`` because it forwards attributes
+    #       from the wrapping function to the wrapped function,
+    #       which will break the code in _make_asserter.
+    @_wraps(fun)
+    def wrapper(*args, **kwargs):
+        result = fun(*args, **kwargs)
+        assert not result, result
```

Comment:
Would it hurt to return the result after the assert?  I get that for your use case here in this file, it doesn't matter, but it might be worth preserving the original function return (even if it is always "false-y") just in case?

Reply:
I am considering this Non blocking.

Reply:
Well, all functions return None if they fall off the bottom, but yes, this deliberately does not return a value just as the `assert` statement (don't even get me started on _that_) does not return a value. It's simply an assert with more helpful semantic messaging - If the code gets passed this point, the assertion was true. :-)